src/Directory.Build.targets

```diff
@@ -132,7 +132,8 @@
 	  macOS needs an explicit override of the current compiler when running under mono.
 	  On other targets, .NET 6.0's compiler takes precedence.
 	  -->
-	  <MicrosoftNetCompilerVersionOverride Condition="'$(MSBuildRuntimeType)'!='Core' and ($([MSBuild]::IsOSPlatform('OSX')) or '$(MSBuildVersion)' &lt; '17.0')">4.0.1</MicrosoftNetCompilerVersionOverride>
+	  <!-- IMPORTANT: TODO: Revert the change I'm doing here before merge. -->
+	  <MicrosoftNetCompilerVersionOverride>4.4.0-1.final</MicrosoftNetCompilerVersionOverride>
```

Comment:
@jeromelaban Is there a problem setting `MicrosoftNetCompilerVersionOverride` unconditionally?

Reply:
Not at all. We can use this version now, if all targets are building properly.